pagefaultgames/pokerogue
# [Bug] Fix reloads causing RNG inconsistencies

## What are the changes the user will see?

Reloads should no longer cause RNG inconsistencies.

## Why am I making these changes?



RNG inconsistencies are not ideal, especially for Daily Runs, where all players should experience the same RNG results. This fix is meant as a quick, temporary fix until Xavion's planned RNG rework.

## What are the changes from a developer perspective?


This "quick fix" simply calls `scene.resetSeed()` at a consistent point at the beginning of each encounter, after the opponent team has been randomly generated, so that the RNG is reset to a consistent state regardless of whether the session was reloaded.
This PR also adds some tools for performing reloads in unit tests.
Also, some further comments:

1. Due to the new RNG resets, some battle mechanics will now share the exact same RNG calls as opponent Pokémon generation, which could lead to issues of the form "Covet is more effective against Species A than Species B". If this is too problematic, we could also try generating the opponent team using `scene.executeWithSeedOffset()` (which isn't ideal, but Xavion's RNG rework will fix this).

2. If anyone knows a more "proper" way to perform reloads in unit tests, please feel free to add it to this PR.

3. (Edit: This PR now also changes Moody and Acupressure to use seeded RNG instead of unseeded RNG.)

## Screenshots/Videos


### Before the changes:

https://github.com/user-attachments/assets/51c70ac5-6a57-4566-b173-de8ae8970fba

Lapras steals 0 items before the reload and then steals 3 items after the reload, which is an unwanted inconsistency.

### After the changes:

https://github.com/user-attachments/assets/bf3e7319-d8c9-44ec-bbde-83235ed5b593

Both before and after the reload, Lapras steals a Sharp Beak and a Liechi Berry on the same hits, which shows RNG consistency.


## How to test the changes?




1. Load a file, and defeat the opponents to progress to the next wave without reloading.
2. Spam Covet/Thief/Dire Claw/another move that calls RNG.
3. Record the RNG results from Step 2.
4. Save and quit, then reload the session.
5. Spam Covet/Thief/Dire Claw/another move that calls RNG.
6. Record the RNG results from Step 5.
7. The RNG results from Steps 3 and 6 should be the same as each other.

## Checklist
